README.md

```diff
@@ -23,104 +23,332 @@
+
+### Pre-render dynamic content (SSG)
+
+The Firebase CLI will detect usage of [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching/get-static-props) and [getStaticPaths](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths).
+
+#### Optional: integrate with Firebase JS SD
+
+When including Firebase JS SDK methods in both server and client bundles, guard against runtime errors by checking isSupported() before using the product. [Not all products are supported in all environments.](https://firebase.google.com/docs/web/environments-js-sdk#other_environments)
+
+Tip: consider using ReactFire, which does this for you automatically.
+
+#### Optional: integrate with the Admin SDK
+
+Admin SDK bundles will fail if included in your browser build;  refer to them only inside [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching/get-static-props) and [getStaticPaths](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths).
+
+### Serve fully dynamic content (SSR)
+
+The Firebase CLI will detect usage of [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props).
+
+### Configure Hosting behavior with next.config.js
+
+#### Image Optimization
+
+Using [Next.js Image Optimization](https://nextjs.org/docs/basic-features/image-optimization) is supported, but it will trigger creation of a Cloud Function, even if you’re not using SSR.
+
+Note: Because of this, Image Optimization and Hosting Preview Channels don’t interoperate well together.
+
+#### Redirects, Rewrites, and Headers
+
+Firebase CLI will respect [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects), [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites), and [headers](https://nextjs.org/docs/api-reference/next.config.js/headers) in your next.config.js, converting them to their respective equivalent Firebase Hosting configuration at deploy time.
+
+If a Next.js redirect, rewrite, or header cannot be converted to an equivalent Firebase Hosting header, it falls back and builds a Cloud Function—even if you aren’t using Image Optimization or SSR.
+
+#### Optional: integrate with Firebase Authentication
+
+The web framework-aware Firebase deployment tooling will automatically keep client and server state in sync using cookies. There are some methods provided for accessing the authentication context in SSR:
+
+* The Express `res.locals` object will optionally contain an authenticated Firebase App instance (`firebaseApp`) and the currently signed in user (`currentUser`).  This can be accessed in `getServerSideProps`.
+* The authenticated Firebase App name is provided on the route query  (__firebaseAppName), this allows for manual integration while in context:
+
+```typescript
+  // get the authenticated Firebase App
+  const firebaseApp = getApp(useRouter().query.__firebaseAppName);
+```
+
+## Guide: “Integrate other frameworks with Express.js”
```

Comment:
We can omit "Guide:"

But actually, are we sure "other frameworks" is in scope for the current plan?

Reply:
Thats a great question!

I don't remember seeing Express / Other frameworks. Maybe this section should get dropped instead?